docs/architecture/adr-059-evidence-composition-and-lifecycle.md

```diff
@@ -0,0 +1,304 @@
+`LightClientAttackEvidence` won't change but the evidence interface will need to look like the proposed one above and `DuplicateVoteEvidence` will need to change to encompass multiple double votes. A problem with batch evidence is that it needs to be unique to avoid people from submitting different permutations.
+
+> I most likely have not captured the full picture of all these approaches and therefore would encourage others to add their takes on each of them.
+
+
+## Evidence Interface
+
+*Note: any changes to the interface will be changing the block data structure and is hence breaking*
+
+As already been alluded to it is most likely that we will need to remove `Verify()` as it's difficult to define a set of arguments that will satisfy the requirements for both duplicate vote verification and light client attack verification (not to mention future evidence and their requirements).
+
+Depending on the composition we might need to change `Address()` to accomodate either individual and/or batch evidence. It may also be a good idea to have internal fields that the evidence pool populates such as time and validators so modules outside of the evidence pool (specifically the `BlockExecutor` don't need to query for more resources)
+
+`Equal()` can also be seen as redundant if we set `Hash()` in the right way that we can just compare evidence hashes for uniqueness. We already use hashes for database lookups
+
+`Type()` could be a new method which would be helpful in constructing abci evidence
+
+`ToABCI()` is another alternative that was written as a TODO in the actual code base.
+
+
+## The State of Amnesia
+
+TBC
+
+## Censorship
+
+A second problem brought up in a prior [issue](https://github.com/tendermint/tendermint/issues/4593) is the notion of evidence censorship. Consider the following scenario: I have greater than 1/3 validator power. I perform an attack either on a light client of on the main network that was detected. Whenever someone proposes a block that contains evidence I abstain from voting meaning that it never is committed. When it is my turn to propose, I propose a block that does not include the evidence, everyone else votes and that block gets committed. Eventually the chain reaches a height and time where the evidence has expired and any attempt to punish my attack has been thwarted.
+
+This would most likely need an RFC (in order to explore possible solutions). As far as I can see there is a 0% approach a 50% approach and 100% approach in terms of the level of intervention.
+
+The 0% approach would just be simply to leave this as is.
+
+The 50% approach would be that validators would consider a proposal invalid if it didn't contain at least one of the evidence that that validator had and thus wouldn't vote on it. This solution is far less intrusive to the current consensus algorithm but it also doesn't quite solve the problem of censorship, rather this protocol will keep the chain halted until off-chain governance can intervene.
```

Comment:
I'm having trouble parsing the first sentence in this paragraph. Can you rephrase?

Reply:
Will do

Reply:
The new phrasing makes perfect sense, thanks!